### PR TITLE
lib: update function definitions to new api

### PR DESCRIPTION
**Pull request by efd6**
This also fixes a compile-time type checking issue.

Unfortunately not mechanical, some colour and movement.

Please take a look.

Notes:

This change has been tested in all integrations that make use of CEL. This requires building a custom filebeat and then injecting that into a running stack. Details below...

```
$ cd $(go env GOPATH)/src/github.com/elastic/beats/x-pack/filebeat
$ go get github.com/elastic/mito@a9f5af2f7468ad8182a9ed73f0b220eabdb2505f
$ go mod tidy
$ PACKAGES=tar.gz PLATFORMS=linux/amd64 DEV=true SNAPSHOT=true mage -v package
```

Extract the zip in place and then...
```
$ docker cp build/distributions/filebeat-8.14.0-SNAPSHOT-linux-x86_64/filebeat <container-id>:/usr/share/elastic-agent/data/elastic-agent-<agent-sha-prefix>/components
$ cd $(go env GOPATH)/src/github.com/elastic/integrations
$ for ds in $(rg -l '^program: \|' | cut -d/ -f1,3 | sort | uniq); do (cd $(dirname $ds); echo $ds; elastic-package test system -d $(basename $ds)); done
```

Output (edited for verbosity)
```
carbon_black_cloud/alert_v7
Run system tests for the package
--- Test results for package: carbon_black_cloud - START ---
╭────────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ carbon_black_cloud │ alert_v7    │ system    │ cel       │ PASS   │ 33.553198415s │
╰────────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: carbon_black_cloud - END   ---
Done
carbon_black_cloud/asset_vulnerability_summary
Run system tests for the package
--- Test results for package: carbon_black_cloud - START ---
╭────────────────────┬─────────────────────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE            │ DATA STREAM                 │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────────────┼─────────────────────────────┼───────────┼───────────┼────────┼───────────────┤
│ carbon_black_cloud │ asset_vulnerability_summary │ system    │ httpjson  │ PASS   │ 31.933203073s │
│ carbon_black_cloud │ asset_vulnerability_summary │ system    │ cel       │ PASS   │ 29.809151676s │
╰────────────────────┴─────────────────────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: carbon_black_cloud - END   ---
Done
carbon_black_cloud/audit
Run system tests for the package
--- Test results for package: carbon_black_cloud - START ---
╭────────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ carbon_black_cloud │ audit       │ system    │ httpjson  │ PASS   │ 31.211807136s │
│ carbon_black_cloud │ audit       │ system    │ cel       │ PASS   │ 29.411471334s │
╰────────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: carbon_black_cloud - END   ---
Done
crowdstrike/alert
Run system tests for the package
--- Test results for package: crowdstrike - START ---
╭─────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE     │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ crowdstrike │ alert       │ system    │ common    │ PASS   │ 37.069836019s │
╰─────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: crowdstrike - END   ---
Done
crowdstrike/host
Run system tests for the package
--- Test results for package: crowdstrike - START ---
╭─────────────┬─────────────┬───────────┬───────────┬────────┬──────────────╮
│ PACKAGE     │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │ TIME ELAPSED │
├─────────────┼─────────────┼───────────┼───────────┼────────┼──────────────┤
│ crowdstrike │ host        │ system    │ common    │ PASS   │ 34.73772644s │
╰─────────────┴─────────────┴───────────┴───────────┴────────┴──────────────╯
--- Test results for package: crowdstrike - END   ---
Done
eset_protect/detection
Run system tests for the package
--- Test results for package: eset_protect - START ---
╭──────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ eset_protect │ detection   │ system    │ default   │ PASS   │ 35.645378942s │
╰──────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: eset_protect - END   ---
Done
eset_protect/device_task
Run system tests for the package
--- Test results for package: eset_protect - START ---
╭──────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ eset_protect │ device_task │ system    │ default   │ PASS   │ 32.108562898s │
╰──────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: eset_protect - END   ---
Done
imperva_cloud_waf/event
Run system tests for the package
--- Test results for package: imperva_cloud_waf - START ---
╭───────────────────┬─────────────┬───────────┬───────────┬────────┬────────────────╮
│ PACKAGE           │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │   TIME ELAPSED │
├───────────────────┼─────────────┼───────────┼───────────┼────────┼────────────────┤
│ imperva_cloud_waf │ event       │ system    │ default   │ PASS   │ 2m7.672016889s │
╰───────────────────┴─────────────┴───────────┴───────────┴────────┴────────────────╯
--- Test results for package: imperva_cloud_waf - END   ---
Done
logstash/node_cel
Run system tests for the package
--- Test results for package: logstash - START ---
╭──────────┬─────────────┬───────────┬────────────────────────────────────┬────────┬───────────────╮
│ PACKAGE  │ DATA STREAM │ TEST TYPE │ TEST NAME                          │ RESULT │  TIME ELAPSED │
├──────────┼─────────────┼───────────┼────────────────────────────────────┼────────┼───────────────┤
│ logstash │ node_cel    │ system    │ default (variant: logstash_8.12.2) │ PASS   │ 41.683114929s │
╰──────────┴─────────────┴───────────┴────────────────────────────────────┴────────┴───────────────╯
--- Test results for package: logstash - END   ---
Done
logstash/pipeline
Run system tests for the package
--- Test results for package: logstash - START ---
╭──────────┬─────────────┬───────────┬────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE  │ DATA STREAM │ TEST TYPE │ TEST NAME                          │ RESULT │ TIME ELAPSED │
├──────────┼─────────────┼───────────┼────────────────────────────────────┼────────┼──────────────┤
│ logstash │ pipeline    │ system    │ default (variant: logstash_8.12.2) │ PASS   │ 43.13600559s │
╰──────────┴─────────────┴───────────┴────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: logstash - END   ---
Done
logstash/plugins
Run system tests for the package
--- Test results for package: logstash - START ---
╭──────────┬─────────────┬───────────┬────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE  │ DATA STREAM │ TEST TYPE │ TEST NAME                          │ RESULT │ TIME ELAPSED │
├──────────┼─────────────┼───────────┼────────────────────────────────────┼────────┼──────────────┤
│ logstash │ plugins     │ system    │ default (variant: logstash_8.12.2) │ PASS   │ 41.97027965s │
╰──────────┴─────────────┴───────────┴────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: logstash - END   ---
Done
menlo/dlp
Run system tests for the package
--- Test results for package: menlo - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬─────────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │    TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼─────────────────┤
│ menlo   │ dlp         │ system    │ default   │ PASS   │ 1m35.505112221s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴─────────────────╯
--- Test results for package: menlo - END   ---
Done
menlo/web
Run system tests for the package
--- Test results for package: menlo - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬─────────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │    TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼─────────────────┤
│ menlo   │ web         │ system    │ default   │ PASS   │ 1m34.517868523s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴─────────────────╯
--- Test results for package: menlo - END   ---
Done
mongodb_atlas/mongod_audit
Run system tests for the package
--- Test results for package: mongodb_atlas - START ---
╭───────────────┬──────────────┬───────────┬───────────┬────────┬─────────────────╮
│ PACKAGE       │ DATA STREAM  │ TEST TYPE │ TEST NAME │ RESULT │    TIME ELAPSED │
├───────────────┼──────────────┼───────────┼───────────┼────────┼─────────────────┤
│ mongodb_atlas │ mongod_audit │ system    │ default   │ PASS   │ 2m41.063346092s │
╰───────────────┴──────────────┴───────────┴───────────┴────────┴─────────────────╯
--- Test results for package: mongodb_atlas - END   ---
Done
mongodb_atlas/process
Run system tests for the package
--- Test results for package: mongodb_atlas - START ---
╭───────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE       │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ mongodb_atlas │ process     │ system    │ default   │ PASS   │ 33.576599878s │
╰───────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: mongodb_atlas - END   ---
Done
o365/audit
Run system tests for the package
--- Test results for package: o365 - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │ TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼──────────────┤
│ o365    │ audit       │ system    │ cel       │ PASS   │ 36.75661057s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴──────────────╯
--- Test results for package: o365 - END   ---
Done
prisma_cloud/alert
Run system tests for the package
--- Test results for package: prisma_cloud - START ---
╭──────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ prisma_cloud │ alert       │ system    │ input     │ PASS   │ 38.427710235s │
╰──────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: prisma_cloud - END   ---
Done
prisma_cloud/audit
Run system tests for the package
--- Test results for package: prisma_cloud - START ---
╭──────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ prisma_cloud │ audit       │ system    │ input     │ PASS   │ 34.344462304s │
╰──────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: prisma_cloud - END   ---
Done
prisma_cloud/host
Run system tests for the package
--- Test results for package: prisma_cloud - START ---
╭──────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ prisma_cloud │ host        │ system    │ udp       │ PASS   │ 40.607040055s │
│ prisma_cloud │ host        │ system    │ input     │ PASS   │ 37.586992525s │
│ prisma_cloud │ host        │ system    │ tcp       │ PASS   │ 41.719486861s │
╰──────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: prisma_cloud - END   ---
Done
prisma_cloud/host_profile
Run system tests for the package
--- Test results for package: prisma_cloud - START ---
╭──────────────┬──────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE      │ DATA STREAM  │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────┼──────────────┼───────────┼───────────┼────────┼───────────────┤
│ prisma_cloud │ host_profile │ system    │ udp       │ PASS   │ 41.283793356s │
│ prisma_cloud │ host_profile │ system    │ input     │ PASS   │ 36.070100403s │
│ prisma_cloud │ host_profile │ system    │ tcp       │ PASS   │  39.01639039s │
╰──────────────┴──────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: prisma_cloud - END   ---
Done
qualys_vmdr/asset_host_detection
Run system tests for the package
--- Test results for package: qualys_vmdr - START ---
╭─────────────┬──────────────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE     │ DATA STREAM          │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────────┼──────────────────────┼───────────┼───────────┼────────┼───────────────┤
│ qualys_vmdr │ asset_host_detection │ system    │ default   │ PASS   │ 36.383668379s │
╰─────────────┴──────────────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: qualys_vmdr - END   ---
Done
qualys_vmdr/knowledge_base
Run system tests for the package
--- Test results for package: qualys_vmdr - START ---
╭─────────────┬────────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE     │ DATA STREAM    │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────────┼────────────────┼───────────┼───────────┼────────┼───────────────┤
│ qualys_vmdr │ knowledge_base │ system    │ 1_invalid │ PASS   │ 35.115765253s │
│ qualys_vmdr │ knowledge_base │ system    │ 0_valid   │ PASS   │ 35.685290683s │
╰─────────────┴────────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: qualys_vmdr - END   ---
Done
symantec_edr_cloud/incident
Run system tests for the package
--- Test results for package: symantec_edr_cloud - START ---
╭────────────────────┬─────────────┬───────────┬───────────┬────────┬──────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │ TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼───────────┼────────┼──────────────┤
│ symantec_edr_cloud │ incident    │ system    │ default   │ PASS   │ 33.68674111s │
╰────────────────────┴─────────────┴───────────┴───────────┴────────┴──────────────╯
--- Test results for package: symantec_edr_cloud - END   ---
Done
ti_crowdstrike/intel
Run system tests for the package
--- Test results for package: ti_crowdstrike - START ---
╭────────────────┬─────────────┬───────────┬───────────┬────────┬──────────────╮
│ PACKAGE        │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │ TIME ELAPSED │
├────────────────┼─────────────┼───────────┼───────────┼────────┼──────────────┤
│ ti_crowdstrike │ intel       │ system    │ default   │ PASS   │ 37.82467619s │
╰────────────────┴─────────────┴───────────┴───────────┴────────┴──────────────╯
--- Test results for package: ti_crowdstrike - END   ---
Done
ti_crowdstrike/ioc
Run system tests for the package
--- Test results for package: ti_crowdstrike - START ---
╭────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE        │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ ti_crowdstrike │ ioc         │ system    │ default   │ PASS   │ 36.728874623s │
╰────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: ti_crowdstrike - END   ---
Done
ti_eclecticiq/threat
Run system tests for the package
--- Test results for package: ti_eclecticiq - START ---
╭───────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE       │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ ti_eclecticiq │ threat      │ system    │ default   │ PASS   │ 37.225964423s │
╰───────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: ti_eclecticiq - END   ---
Done
ti_opencti/indicator
Run system tests for the package
--- Test results for package: ti_opencti - START ---
╭────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ ti_opencti │ indicator   │ system    │ default   │ PASS   │ 36.778168017s │
╰────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: ti_opencti - END   ---
Done
ti_otx/pulses_subscribed
Run system tests for the package
--- Test results for package: ti_otx - START ---
╭─────────┬───────────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM       │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼───────────────────┼───────────┼───────────┼────────┼───────────────┤
│ ti_otx  │ pulses_subscribed │ system    │ default   │ PASS   │ 35.545566645s │
╰─────────┴───────────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: ti_otx - END   ---
Done
ti_threatconnect/indicator
Run system tests for the package
--- Test results for package: ti_threatconnect - START ---
╭──────────────────┬─────────────┬───────────┬───────────┬────────┬──────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │ TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼───────────┼────────┼──────────────┤
│ ti_threatconnect │ indicator   │ system    │ default   │ PASS   │ 36.34321235s │
╰──────────────────┴─────────────┴───────────┴───────────┴────────┴──────────────╯
--- Test results for package: ti_threatconnect - END   ---
Done
trellix_epo_cloud/device
Run system tests for the package
--- Test results for package: trellix_epo_cloud - START ---
╭───────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE           │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ trellix_epo_cloud │ device      │ system    │ default   │ PASS   │ 34.019966923s │
╰───────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: trellix_epo_cloud - END   ---
Done
trellix_epo_cloud/event
Run system tests for the package
--- Test results for package: trellix_epo_cloud - START ---
╭───────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE           │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ trellix_epo_cloud │ event       │ system    │ default   │ PASS   │ 34.002782312s │
╰───────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: trellix_epo_cloud - END   ---
Done
trellix_epo_cloud/group
Run system tests for the package
--- Test results for package: trellix_epo_cloud - START ---
╭───────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE           │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ trellix_epo_cloud │ group       │ system    │ default   │ PASS   │ 34.606420333s │
╰───────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: trellix_epo_cloud - END   ---
Done
wiz/audit
Run system tests for the package
--- Test results for package: wiz - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ wiz     │ audit       │ system    │ default   │ PASS   │ 34.654596102s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: wiz - END   ---
Done
wiz/issue
Run system tests for the package
--- Test results for package: wiz - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ wiz     │ issue       │ system    │ default   │ PASS   │ 35.352024895s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: wiz - END   ---
Done
wiz/vulnerability
Run system tests for the package
--- Test results for package: wiz - START ---
╭─────────┬───────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM   │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼───────────────┼───────────┼───────────┼────────┼───────────────┤
│ wiz     │ vulnerability │ system    │ default   │ PASS   │ 35.345787921s │
╰─────────┴───────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: wiz - END   ---
Done
```

Sanity check
```
$ cd $(go env GOPATH)/src/github.com/elastic/beats/x-pack/filebeat
$ md5sum build/distributions/filebeat-8.14.0-SNAPSHOT-linux-x86_64/filebeat
6db9d666f528ffbf2087be42c74d9b23  …/github.com/elastic/beats/x-pack/filebeat/build/distributions/filebeat-8.14.0-SNAPSHOT-linux-x86_64/filebeat
$ docker exec 14e70ce88e33d7dc32c439194492813d7df5f2f755cd1a6046c8aee1f74e086e md5sum /usr/share/elastic-agent/data/elastic-agent-1eb18c/components/filebeat
6db9d666f528ffbf2087be42c74d9b23  /usr/share/elastic-agent/data/elastic-agent-1eb18c/components/filebeat
```

### CODE REVIEW

File: lib/json.go [hunk 1]
```diff
@@ -79,131 +76,89 @@ import (
 //
 //	'{"a":1}{"b":2}'.decode_json_stream()   // return [{"a":1}, {"b":2}]
 //	b'{"a":1}{"b":2}'.decode_json_stream()  // return [{"a":1}, {"b":2}]
-func JSON(adapter ref.TypeAdapter) cel.EnvOption {
+func JSON(adapter types.Adapter) cel.EnvOption {
```
Comment (efd6) [inline lib/json.go:79]: This is not a breaking change as `types.Adapter` is [`type Adapter = ref.TypeAdapter`](https://pkg.go.dev/github.com/google/cel-go@v0.20.1/common/types#Adapter).

File: lib/xml.go [hunk 2]
```diff
@@ -61,7 +58,7 @@ import (
 //	b"<?xml vers... ...>".decode_xml()   // return { ... }
 //	"<?xml vers... ...>".decode_xml("xsd")   // return { ... }
 //	b"<?xml vers... ...>".decode_xml("xsd")   // return { ... }
-func XML(adapter ref.TypeAdapter, xsd map[string]string) (cel.EnvOption, error) {
+func XML(adapter types.Adapter, xsd map[string]string) (cel.EnvOption, error) {
```
Comment (efd6) [inline lib/xml.go:61]: This is not a breaking change as `types.Adapter` is [`type Adapter = ref.TypeAdapter`](https://pkg.go.dev/github.com/google/cel-go@v0.20.1/common/types#Adapter).

File: lib/collections.go [hunk 3]
```diff
@@ -270,293 +267,191 @@ type collectionsLib struct{}
...
+				"zip_list",
+				[]*cel.Type{listK, listV},
+				mapKV,
+				cel.BinaryBinding(zipLists),
+			),
 		),
 	}
 }
 
+func (collectionsLib) ProgramOptions() []cel.ProgramOption { return nil }
```
Comment (kcreddy) [inline lib/collections.go:453]: I believe this function definition is still required and cannot be omitted?

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

File: lib/collections.go (hunk 3 above)
Comment (efd6) [reply lib/collections.go:453]: That's correct. This is required for Go interface satisfaction.

**Pull request merged**